test_util/launch.py

```diff
@@ -1,20 +1,169 @@
-"""
+"""DC/OS Launch
 
 Usage:
-dcos-launch create [--wait] [--dump-info=cluster_info.json] <config=config.yaml>
-dcos-launch wait <path=cluster_info.json>
-dcos-launch describe <path=cluster_info.json>
-dcos-launch delete <path=cluster_info.json>
+  dcos-launch create [--config-path=<path>] [--info-path=<path>]
+  dcos-launch wait [--info-path=<path>]
+  dcos-launch describe [--info-path=<path>]
+  dcos-launch delete [--info-path=<path>]
+
+Commands:
+  create    reads the file given by --config-path, creates the cluster described therein
+            and finally dumps a JSON file to the path given in --info-path which can then
+            be used with the wait, describe, and delete calls
+  wait      Block until the cluster is up and running
+  describe  Return additional information about the composition of the cluster
+  delete    Destroying the provided cluster deployment
+
+Options:
+  --config-path=<path>  Path for config to create cluster from [default: config.yaml].
+  --info-path=<path>    JSON file output by create and consumed by wait, describe,
+                        and delete [default: cluster_info.json].
 """
+import abc
+import copy
+import os
+import sys
 
+import yaml
 from docopt import docopt
 
+from pkgpanda.util import json_prettyprint, load_json, load_string, write_json
+
+from test_util.aws import BotoWrapper, DcosCfSimple
+
+
+class LauncherError(Exception):
+    def __init__(self, error, msg):
+        self.error = error
+        self.msg = msg
+
+    def __repr__(self):
+        return '{}: {}'.format(self.error, self.msg)
+
+
+class AbstractLauncher(metaclass=abc.ABCMeta):
+    def create(self, config):
+        raise NotImplementedError()
```

Comment:
The dcos/dcos codebase currently has 14 instances of throwing NotImplementedError, and all of those use () for actually constructing an instance of the class, rather than throwing the exception class itself.